src/intent.html

```diff
@@ -245,9 +247,19 @@ <h3 id="mixing_intent_dictionaries">Intent Concept Dictionaries</h3>
      <a data-cite="INFRA#ascii-case-insensitive">ASCII case-insensitive</a>
      and also normalize
      <q>`_`</q> (U+00F5) and <q>`.`</q>  (U+002E) to <q>`-`</q> (U+002D).
-     If the speech hints are not being used
-     and the concept name is being read then each of  `-`, `_` and `.` should be
-     read as an inter-word space.</p>
+     An intent matches an entry in the core or open concept list when the normalized name, the fixity property
+     (which may be defaulted in the concept dictionary), and the number of arguments all match.
+     If a match is found, the speech hint in the list
+     should be used be used as a guide for the generation of suitable text
+     for the <a href="#intent_known_concept">known concept</a>.
+     If the intent does not match any entries, then it is treated as
+     an <a href="#intent_unknown_concept">unknown concept</a>.
```

Comment:
I think Neil's wording is better. A concept not matching the lists known to the system is literally the definition of `unknown-concept` so the fact that these are treated as unknown concepts is by definition not just an expected behaviour.

```
An <dfn id="intent_unknown_concept">unknown concept</dfn> is a concept not
      currently known to the AT. These will be treated the same as a [=literal=], spoken as-is.
      However, future updates of the AT and [=Intent Concept Dictionary=] may
      include additional concepts, at which time those concepts may also receive special treatment
```

Reply:
There is a subtle difference:
1. "then it is treated as" **prescribes** the 'unknown concept' behavior. 
2. "the expected baseline behavior is" signals AT can **choose to do more**.

The subtle difference is also in the text you quoted - "not currently known to the AT" is implementation-specific, and is in no way equivalent to "not in the Core or Open lists".

My two old examples where this is important were:
- a hypothetical ChemistCat AT which has its own (Open-like) list of chemistry concepts and speech hints. 
- a ML (machine learning) system which does not employ explicit lists, but is trained on a large Open domain of intent values.

I am (and have been) intentionally attempting to get a different emphasis across in the wording here.

Reply:
> a hypothetical ChemistCat AT which has its own (Open-like) list of chemistry concepts and speech hints.

That is more or less a canoncal example of a system specific list, nothing here in any way suggests you can not do that.

    a ML (machine learning) system which does not employ explicit lists, but is trained on a large Open domain of intent values.

We had similar issues with openmath content dictionaries that were more or less documented as finite lists (of xml elements in that case) but as they could be (and were) generated on demand by some systems, they could allow a lot more than a first reading might suggest.

Your system could (depending on exactly what it does) have a concept dictionary that matches every NCName. If that's what you document that it does, then you match any name (and arity) and make whatever speech you want.

Reply:
I am not sure I can restate this any more cleanly, but let me try again:

If a concept does not match _a list's speech hints_ (= the situation described by the current text), that is not the same as the concept not being _known to an AT_ (an "unknown concept"). To coin a clarifying term, we are discussing an "unmatched concept", not an "unknown" one.

The definition of "unknown concept" is not directly applicable here and **prescribing that it is** is a design choice.

And I object to that choice, because I want to protect the open experimental "gray area" between intent uses clearly prescribed in lists, and intent uses that are entirely unknown to an AT system. The space in between needs to be inhabited by experiments, so that the lists (especially the Open lists) can continue to grow when experiments stabilize.

Reply:
> If a concept does not match a list's speech hints (= the situation described by the current text), that is not the same as the concept not being known to an AT (an "unknown concept"). To coin a clarifying term, we are discussing an "unmatched concept", not an "unknown" one.

Yes they are the same. An unknown concept is (by definition) a name not in the dictionary supported by the AT in use.
Such a concept is called an unknown concept and the behavior for unknown concepts is prescribed as reading them as literals.

If you have a AT system with a dynamic concept dictionary that matches every NCName then you have no unknown concepts and can do prettty much anything with any concept name.

Reply:
> An unknown concept is (by definition) a name not in the dictionary supported by the AT in use.

Not the current definition, which is one I quite like: `"not currently known to the AT"`

An example I've given before (in a discussion with Moritz) is that if an AT implements a small subset of the Open list -- and only that -- any concept it hasn't yet covered would be an "unknown concept" to the AT, despite being listed. Which is _why_ we could expect the "read as-is" graceful fallback behavior in partial implementations of any list.

"unknown concept" has operational semantics _internal to an AT_, and is not a global list-related property.

I do not understand the dogmatic defense of the prescriptive text here. My change keeps the baseline behavior exactly the same, it only leaves the door open that an AT can do more when desired.

Reply:
> Not the current definition, which is one I quite like: "not currently known to the AT"
yes, by that definition.

> any concept it hasn't yet covered would be an "unknown concept" to the AT, despite being listed. 

what is listed in the open list has no bearing on what is an unknown concept. It is as I said, just a list of suggestions to implementors should they wish to make anything known.

An unknown concept is a concept not in the list of names _implemented by the system in use_  Most of the names in the open list are unknown concepts in most systems. Despite the fact that they are of course listed in the open list.

> I do not understand the dogmatic defense of the prescriptive text here. 

Sorry I think you have simply misread the definition there is no prescriptive text here. I do not think that the suggested text makes sense in a specification context.

> When an intent does not match any entries in AT-supported lists, the expected baseline behavior is that for an unknown concept

the premise of this statement is exactly the definition of unknown concept so this means

> for an unknown concept the expected baseline behaviour is that of an unknown concept.

This is tautologous and even if it were not "expected baseline behaviour" is an undefined term (I could guess a meaning) but it is not a phrase suitable for a spcification.

Reply:
Some of this discussion is a little hard to track, and seems to lead to distributed definitions.  I think that where we have `<dfn id="intent_known_concept">known concept</dfn>` (currently top of the page), as much as possible we should have a fairly clear and complete definition. It might be worth making it more explicit that "known" means the *union* of Core, Open *and* any other implicit dictionaries supported by the AT in question and is thus system specific.

Reply:
> It might be worth making it more explicit that "known" means the union of Core, Open and any other implicit dictionaries supported by the AT in question and is thus system specific.

As currently defined it is _just_ those supported by the system not the core or open lists, the complement  to unknown which is defined in the current daft as:

> unknown concept is a concept not currently known to the AT

This list _should_ include all of core, but if you are using a system that does not implement all the core list then some core concepts could also be unknown.

a name being in the open list does not affect whether it is known or unknown in that sense. It is just a list of suggestions to system implementors for concepts they may want to implement.

Reply:
Ah, yes, right; good point. But then, the section that sparked this discussion `An intent matches an entry in the core or open concept list when ...` shouldn't be mentioning the open or core lists at all, but rather should refer to the AT's own concept list (whatever it may be).

Reply:
I think that works, yes.